Sources/Starknet/Data/TypedData/StarknetTypedData.swift

```diff
@@ -296,30 +296,33 @@ public struct StarknetTypedData: Codable, Equatable, Hashable {
             return hashArray(hashes)
         }
 
-        switch (typeName, revision) {
-        case ("felt", _), ("string", .v0), ("shortstring", .v1), ("ContractAddress", .v1), ("ClassHash", .v1):
+        let basicType = BasicType(rawValue: typeName)
+        switch (basicType, revision) {
+        case (.felt, _), (.string, .v0), (.shortstring, .v1), (.contractAddress, .v1), (.classHash, .v1):
             return try unwrapFelt(from: element)
-        case ("u128", .v1), ("timestamp", .v1):
+        case (.u128, .v1), (.timestamp, .v1):
             return try unwrapU128(from: element)
-        case ("i128", .v1):
+        case (.i128, .v1):
             return try unwrapI128(from: element)
-        case ("bool", _):
+        case (.bool, _):
             return try unwrapBool(from: element)
-        case ("string", .v1):
+        case (.string, .v1):
             return try hashArray(unwrapLongString(from: element))
-        case ("selector", _):
+        case (.selector, _):
             return try unwrapSelector(from: element)
-        case ("enum", .v1):
+        case (.enum, .v1):
             guard let context else {
                 throw StarknetTypedDataError.contextNotDefined
             }
             return try unwrapEnum(from: element, context: context)
-        case ("merkletree", _):
+        case (.merkletree, _):
             guard let context else {
                 throw StarknetTypedDataError.contextNotDefined
             }
             return try prepareMerkleTreeRoot(from: element, context: context)
-        default:
+        case (.some, .v0):
+            throw StarknetTypedDataError.unsupportedType(typeName)
```

Comment:
Not sure whether we should throw `dependencyNotDefined` or `unsupportedType` here. Technically `dependencyNotDefined` is accurate for the legacy version, but for the most users I think `unsupportedType` provides more info.

Reply:
InsupportedType makes sense here, I'd just leave it as is